packages/flutter_tools/test/general.shard/isolated/macos/native_assets_test.dart

```diff
@@ -280,30 +287,129 @@ void main() {
                 'lipo',
                 '-create',
                 '-output',
-                dylibPath,
+                dylibPathBar,
                 'arm64/libbar.dylib',
                 'x64/libbar.dylib',
               ],
             ),
-            if  (!flutterTester)
+            if (!flutterTester)
+              FakeCommand(
+                command: <Pattern>[
+                  'otool',
+                  '-D',
+                  dylibPathBar,
+                ],
+                stdout: <String>[
+                  '$dylibPathBar (architecture x86_64):',
+                  '@rpath/libbar.dylib',
+                  '$dylibPathBar (architecture arm64):',
+                  '@rpath/libbar.dylib',
+                ].join('\n'),
+              ),
+            if (!flutterTester)
               FakeCommand(
                 command: <Pattern>[
                   'install_name_tool',
                   '-id',
                   '@rpath/bar.framework/bar',
-                  dylibPath,
+                  dylibPathBar,
+                ],
+              ),
+            if (flutterTester)
+              FakeCommand(
+                command: <Pattern>[
+                  'codesign',
+                  '--force',
+                  '--sign',
+                  '-',
+                  if (buildMode == BuildMode.debug) '--timestamp=none',
+                  signPathBar,
                 ],
               ),
             FakeCommand(
               command: <Pattern>[
-                'codesign',
-                '--force',
-                '--sign',
-                '-',
-                if (buildMode == BuildMode.debug) '--timestamp=none',
-                signPath,
+                'lipo',
+                '-create',
+                '-output',
+                dylibPathBuz,
+                'arm64/libbuz.dylib',
+                'x64/libbuz.dylib',
               ],
             ),
+            if (!flutterTester)
+              FakeCommand(
+                command: <Pattern>[
+                  'otool',
+                  '-D',
+                  dylibPathBuz,
+                ],
+                stdout: <String>[
+                  '$dylibPathBuz (architecture x86_64):',
+                  '@rpath/libbuz.dylib',
+                  '$dylibPathBuz (architecture arm64):',
+                  '@rpath/libbuz.dylib',
+                ].join('\n'),
+              ),
+            if (!flutterTester)
+              FakeCommand(
+                command: <Pattern>[
+                  'install_name_tool',
+                  '-id',
+                  '@rpath/buz.framework/buz',
+                  dylibPathBuz,
+                ],
+              ),
+            if (flutterTester)
```

Comment:
`else` (and maybe invert order to get rid of `!`)

Or rewrite completely to `if (flutterTester) ... [` ... `] else ... [` ... `]`.